### PR TITLE
Update to Solr 9.3

### PR DESCRIPTION
**Pull request by lfarrell**
Update to Solr 9.3 and completely update the DataImportHandler as it's no longer included with Solr.
This PR pulls in and uses https://github.com/SearchScale/dataimporthandler for data import handling.

https://unclibrary.atlassian.net/browse/APPDEV-10900

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -55,4 +55,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
         config.hostmanager.manage_host = true
         config.hostmanager.aliases = settings['sites'].map { |site| site['map'] }
     end
+
+    # Solr
+    config.vm.network :forwarded_port, guest: 8983, host: 8983
```
Comment (sharonluong) [inline Vagrantfile:60]: oh! this makes sense. thanks for adding it

Comment (lfarrell) [reply Vagrantfile:60]: I got tired of having to add the port every time I rebooted the VM :)

**Comment by sharonluong** [comment]
My tests seem to be going well! I think the only thing is that the readme needs to be updated, and then I'll approve.

**Review by sharonluong — APPROVED** [review]
Works great! thanks!

**Pull request merged**